src/graphics/resources/texture.rs

```diff
@@ -93,18 +104,22 @@ fn view(texture: &wgpu::Texture) -> wgpu::TextureView {
     texture.create_view(&wgpu::TextureViewDescriptor::default())
 }
 
-fn bytes_per_row(cfg: &TextureConfig) -> u32 {
+/// Returns the number of bytes per row and the padding width
+fn bytes_layout(cfg: &TextureConfig) -> (u32, u32) {
     // Always returns `Some(u32)` when using `Rgba8Unorm`
     let block_size = cfg
         .format
         .block_copy_size(None)
         .expect("Bad texture format");
 
-    let bytes_size = block_size * cfg.size.width;
+    let width = block_size * cfg.size.width;
 
     match cfg.pad_bytes_per_row {
-        true => pad_size(bytes_size, 256),
-        false => bytes_size,
+        true => {
```

Comment:
This is more of a question about programming style : why do you use a match for a boolean result? In what way(s) is it better (or worse) than an `if`?

Reply:
I do it only because it saves one line when using `rustfmt` and I find it easier on the eyes. I know this is not a very good reason. 🤷